src/generate.js

```diff
@@ -258,81 +270,71 @@ export const generateCSSRuleset = (
+                    originalStyle = elementNames[i][3].toLowerCase() + elementNames[i].slice(4);
+                } else { // if (elementNames[i][1] === 's') {
+                    // This is a Ms-prefixed style, like "MsTransition".
+                    originalStyle = elementNames[i][2].toLowerCase() + elementNames[i].slice(3);
+                }
 
-    // Calculate the sort order of a given property.
-    function sortOrderForProperty([key, value]) {
-        if (sortOrder.hasOwnProperty(key)) {
-            if (originalElements.hasOwnProperty(key) &&
-                    originalElements[key] !== value) {
-                // The value is prefixed. Sort this just before the key with
-                // the unprefixed value.
-                return sortOrder[key] - 0.25;
-            } else {
-                // Either the key and value are unprefixed here, or this is a
-                // prefixed key. Either way, this is handled by the sortOrder
-                // calculation above.
-                return sortOrder[key];
+                if (originalStyle && originalElements.hasOwnProperty(originalStyle)) {
+                    const originalIndex = handledDeclarations.keyOrder.indexOf(originalStyle);
+                    handledDeclarations.keyOrder.splice(originalIndex, 0, elementNames[i]);
+                } else {
+                    // We don't know what the original style was, so sort it to
+                    // top. This can happen for styles that are added that don't
+                    // have the same base name as the original style.
+                    handledDeclarations.keyOrder.unshift(elementNames[i]);
+                }
             }
-        } else {
-            // If the property isn't in the sort order, it wasn't in the
-            // original set of unprefixed keys, so it must be a prefixed key.
-            // Sort at order -1 to put it at the top of the set of styles.
-            return -1;
         }
     }
 
-    // Actually sort the rules according to the sort order.
-    prefixedRules.sort(
-        (a, b) => sortOrderForProperty(a) - sortOrderForProperty(b));
-
     const transformValue = (useImportant === false)
         ? stringifyValue
         : (key, value) => importantify(stringifyValue(key, value));
 
-    const rules = prefixedRules
-        .map(([key, value]) => `${kebabifyStyleName(key)}:${transformValue(key, value)};`)
-        .join("");
+    const rules = [];
+    for (let i = 0; i < handledDeclarations.keyOrder.length; i ++) {
+        const key = handledDeclarations.keyOrder[i];
+        const value = prefixedElements[key];
+        if (Array.isArray(value)) {
+            // inline-style-prefixer returns an array when there should be
+            // multiple rules for the same key. Here we flatten to multiple
+            // pairs with the same key.
+            for (let j = 0; j < value.length; j++) {
+                rules.push(transformRule(key, value[j], transformValue));
```

Comment:
I think that's correct. At least in the examples I was looking at, this seemed to be the order we wanted. Although I'm not sure if that is always true or if it will remain stable. It might be worth adding a check here to ensure that the original or unprefixed value is last.